microgen/remesh.py

```diff
@@ -1,64 +1,117 @@
 import os
-import subprocess
 from tempfile import NamedTemporaryFile
-from typing import List
+from typing import List, Optional, Union
 
 import pyvista as pv
 
-from microgen import BoxMesh
+from microgen import BoxMesh, Mmg, is_periodic
+
+
+class InputMeshNotPeriodicError(Exception):
+    """Raised when input mesh of remesh_keeping_periodicity_for_fem is not periodic"""
+
+
+class OutputMeshNotPeriodicError(Exception):
+    """Raised when output mesh of remesh_keeping_periodicity_for_fem is not periodic"""
 
 
 def remesh_keeping_periodicity_for_fem(
-    input_mesh: BoxMesh,
-    output_mesh_file: str,
+    input_mesh: Union[BoxMesh, pv.UnstructuredGrid, pv.PolyData],
     mesh_version: int = 2,
     dimension: int = 3,
-    hausd: float = None,
-    hgrad: float = None,
-    hmax: float = None,
-    hmin: float = None,
-    hsiz: float = None,
-) -> None:
+    tol: float = 1e-8,
+    hausd: Optional[float] = None,
+    hgrad: Optional[float] = None,
+    hmax: Optional[float] = None,
+    hmin: Optional[float] = None,
+    hsiz: Optional[float] = None,
+) -> pv.UnstructuredGrid:

```

Comment:
Why the returned mesh must be necessarily a pv.UnstructuredGrid ? It should not depend on the input mesh type ?

Reply:
Good remark. I have now added the possibility to have a `BoxMesh` or `pv.UnstructuredGrid` output based on input type, and overloaded the function